test/src/constraints/GapHeatConductanceTest.C

```diff
@@ -64,8 +71,16 @@ GapHeatConductanceTest::computeQpResidual(Moose::MortarType type)
       heat_transfer_coeff =
           (0.5 * (_secondary_gap_conductance[_qp] + _primary_gap_conductance[_qp])) / gap;
 
-      return _test[_i][_qp] *
-             (_lambda[_qp] - heat_transfer_coeff * (_u_secondary[_qp] - _u_primary[_qp]));
+      const auto u_secondary =
+          _functor_evals_for_primal
+              ? _secondary_var(
+                    ElemPointArg({_interior_secondary_elem, _phys_points_secondary[_qp], false}))
+              : _u_secondary[_qp];
+      const auto u_primary = _functor_evals_for_primal
+                                 ? _primary_var(ElemPointArg(
+                                       {_interior_primary_elem, _phys_points_primary[_qp], false}))
+                                 : _u_primary[_qp];
+      return _test[_i][_qp] * (_lambda[_qp] - heat_transfer_coeff * (u_secondary - u_primary));
```

Comment:
I take it that other gap heat transfer objects can be expanded to work with second-order mortar FV by simply accounting for the difference in evaluation of primal variables.

Reply:
yes. I think we can create some new gap flux models in `heat_conduction` that account for the different flux computation methods